thread.c

```diff
@@ -2720,7 +2751,10 @@ rb_thread_io_close_interrupt(struct rb_io *io)
     // This is used to ensure the correct execution context is woken up after the blocking operation is interrupted:
     io->wakeup_mutex = rb_mutex_new();
 
-    return thread_io_close_notify_all(io);
+    // We need to use a mutex here as entering the fiber scheduler may cause a context switch:
```

Comment:
it means "without fiber scheduler, the mutex is not needed".
Could you introduce the check?

Reply:
That won’t work. It’s not about the current thread scheduler, but the scheduler on the thread we are interrupting.